lib/crypto/c_src/cipher.c

```diff
@@ -28,98 +28,192 @@
 
 static struct cipher_type_t cipher_types[] =
 {
-    {{"rc2_cbc"},
 #ifndef OPENSSL_NO_RC2
-     {&EVP_rc2_cbc}
+    {{"rc2_cbc"}, {&EVP_rc2_cbc}, 0, NO_FIPS_CIPHER},
 #else
-     {NULL}
+    {{"rc2_cbc"}, {NULL}, 0, NO_FIPS_CIPHER},
 #endif
-    ,0},
-    {{"des_cbc"}, {COND_NO_DES_PTR(&EVP_des_cbc)}, 0},
-    {{"des_cfb"}, {COND_NO_DES_PTR(&EVP_des_cfb8)}, 0},
-    {{"des_ecb"}, {COND_NO_DES_PTR(&EVP_des_ecb)}, 0},
-    {{"des_ede3_cbc"}, {COND_NO_DES_PTR(&EVP_des_ede3_cbc)}, 0},
-    {{"des_ede3_cbf"}, /* Misspelled, retained */
+    {{"rc4"},     {&EVP_rc4}, 0, NO_FIPS_CIPHER},
```

Comment:
Seems like this should be wrapped in a `#ifndef OPENSSL_NO_RC4` since that define exists in OpenSSL.